File: src/Page/SelectableCoursesPage.java
package Page;

import SystemMessage.FunctionType;
import SystemMessage.SystemMessage;

public class SelectableCoursesPage extends Page{
	
	private int numberOfSelectableCourses;
	
	public SelectableCoursesPage(String content) {
		super(content);
		setType(PageType.SELECTABLE_COURSES_PAGE);
		setName("Selectable Courses Page");
	}
	
	@Override
	public SystemMessage runPage() {
		showContent();
		String input;
		while(true) {		
			input = takeInput();
			try {
				int selection = Integer.parseInt(input);
				if (selection > numberOfSelectableCourses || selection < 0) {
					System.out.println("Enter a valid number");
				}
				else {
					return new SystemMessage(FunctionType.SELECT_COURSE, PageType.SELECTABLE_COURSES_PAGE, selection);
				}
			} 
			catch (NumberFormatException e) {
				if (input.equalsIgnoreCase("q")) {
					return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.MAIN_MENU_PAGE_STUDENT, null);
				}
				else {
					System.out.println("Wrong Input");
				}
			}	
		}		
	}
	
	/////////// GETTER - SETTER METHODS ////////////
	public int getNumberOfSelectableCourses() {
		return numberOfSelectableCourses;
	}

	public void setNumberOfSelectableCourses(int numberOfSelectableCourses) {
		this.numberOfSelectableCourses = numberOfSelectableCourses;
	}
}

File: python_development/Page/SelectableCoursesPage.py
from Page import Page
from Page import PageType
from System import FunctionType
from System import SystemMessage

class SelectableCoursesPage(Page):

    numberOfSelectableCourses = None

    def __init__(self, content):
        super().__init__(content)
        self.setType(PageType.SelectableCoursesPage)
        self.setName("Selectable Courses Page")


    def runPage(self):
        self.showContent()
        inputFromPage = None
        while(True):
            inputFromPage = self.takeInput()
            try:
                selection = int(inputFromPage)
                if (selection > self.numberOfSelectableCourses or selection < 0):
                    print("Enter a valid number")
                else: 
                    return SystemMessage(FunctionType.SELECT_COURSE, PageType.SELECTABLE_COURSES_PAGE, selection)
            except:
                if (inputFromPage.lower() == "q"):
                    return SystemMessage(FunctionType.CHANGE_PAGE, PageType.MAIN_MENU_PAGE, None)
                else:
                    print("Wrong Input!")


    def getNumberOfSelectableCourses(self):
        return self.numberOfSelectableCourses
    
    def setNumberOfSelectableCourses(self, numberOfSelectableCourses):
        self.numberOfSelectableCourses = numberOfSelectableCourses